### DISCUSSION TITLE
Getting error when using smtp-relay.gmail.com in pocketbase

### DISCUSSION THREAD

**Discussion by gaganbiswas** [pocketbase/pocketbase#3097] [Q&A]
I have set up my app password as mentioned by @ganigeorgiev in another forum and used the same in the mail settings of pocketbase. I am getting an EOF error while trying to send test mails. What is going wrong here? 

Instead of using smtp.gmail.com (this works). I am using smtp-relay.gmail.com.

**Comment by ganigeorgiev** [comment]
I remember testing the steps from https://github.com/pocketbase/pocketbase/discussions/458#discussioncomment-3651995, but it's been a while and it is possible that gmail may have changed something.

Could you post the exact error? Why do you need to use `smtp-relay.gmail.com` instead of `smtp.gmail.com`? Is it for rerouting outgoing emails? If it is for rerouting unfortunately I'm not familiar with this setup and I'm not sure whether there are any extra requirements that the client/domain need to satisfy.

**Reply by ganigeorgiev** [reply]
_If you haven't read it already, this guide could be helpful - https://support.google.com/a/answer/2956491_

**Reply by gaganbiswas** [reply]
According to google, if someone is sending mail outside of the organization account from an external source like app or device, then it is recommended to use `smtp-relay.gmail.com` rather than `smtp.google.com` https://support.google.com/a/answer/176600?hl=en. It is most secure and also the sending limit is 10,000 recipients per day where the later has a limit of 2000.

`smtp.gmail.com` works fine for me as of now. If you get time in the process of development kindly look into the requirements of smtp-relay.gmail.com

**Reply by gaganbiswas** [reply]
> _If you haven't read it already, this guide could be helpful - https://support.google.com/a/answer/2956491_

I am already using it in a bulk mailer using python that I created and it's working fine. In pocketbase I am getting an error `Failed to send test email. Raw Error: EOF`.

**Reply by ganigeorgiev** [reply]
Are you sending from the same domain and server? You can also try to start the executable with `--debug` to see if there are any more detailed errors logged in the console/stderr.

I don't think I can help you debug it further because it seems that you need a workspace/enterprise account to use the relay gmail service. I have no idea why it would work with python and not with Go if the same configuration and domain setup is used.

Googling for the smtp-relay, I also found this post - https://www.cubebackup.com/blog/how-to-use-google-smtp-service-to-send-emails-for-free/ (look for _"Method 3:  Using Google Workspace SMTP relay service."_), that contains detailed steps how to enable and configure the relay account.

**Reply by gaganbiswas** [reply]
> Are you sending from the same domain and server? You can also try to start the executable with `--debug` to see if there are any more detailed errors logged in the console/stderr.
> 
> I don't think I can help you debug it further because it seems that you need a workspace/enterprise account to use the relay gmail service. I have no idea why it would work with python and not with Go if the same configuration and domain setup is used.

> Googling for the smtp-relay, I also found this post - https://www.cubebackup.com/blog/how-to-use-google-smtp-service-to-send-emails-for-free/ (look for _"Method 3: Using Google Workspace SMTP relay service."_), that contains detailed steps how to enable and configure the relay account.

All the steps from the above blog are already being followed, since i am already using it in my python bulk mailer app. I will look into the debugging part and if I get something will let you know. Thank your for your quick replies 😄 . I will let you know once I get something useful from the debugger.

**Reply by ganigeorgiev** [reply]
I think this is caused by the EHLO/HELO configuration because we use localhost as noted in https://github.com/pocketbase/pocketbase/discussions/458#discussioncomment-3651995 and https://github.com/pocketbase/pocketbase/discussions/439#discussioncomment-3651399, which when using the gmail smtp-relay fails because it seems to require a proper domain name.

I don't have an account to confirm that this is the case but I found a similar issue in the mail library we use - https://github.com/domodwyer/mailyak/pull/79. I'll consider adding a mailer config option with the next release to specify an explicit local sender part.

**Reply by ganigeorgiev** [reply]
@gaganbiswas I've added in the `develop` branch (https://github.com/pocketbase/pocketbase/commit/b2ac538580ddf88829f3f53e5b2f80a117ffc209) a new "EHLO/HELO domain" field and it will be part of v0.18.0 (_there are no ETAs yet_).

I don't have access to paid workspace/enterprise account to test the Gmail SMTP-relay, but based on https://github.com/domodwyer/mailyak/pull/79, I suspect that this is the same case.

**Reply by gaganbiswas** [reply]
Thank you for considering the issue. Will be waiting for the v0.18.0 release. Meanwhile if I get my hands on any workspace account I will be sharing the same with you for helping you out to test the same.
